velox/vector/arrow/Bridge.cpp

```diff
@@ -777,18 +780,18 @@ void exportViews(
       stringBufferVec.begin(),
       stringBufferVec.end(),
       [&out](const auto& lhs, const auto& rhs) {
-        return ((uint64_t*)&out.buffers[2])[lhs] <
-            ((uint64_t*)&out.buffers[2])[rhs];
+        return reinterpret_cast<uint64_t*>(&out.buffers[2])[lhs] <
+            reinterpret_cast<uint64_t*>(&out.buffers[2])[rhs];
       });
 
-  auto utf8Views = (uint64_t*)out.buffers[1];
+  auto utf8Views = reinterpret_cast<const uint64_t*>(out.buffers[1]);
   int32_t bufferIdxCache = 0;
   uint64_t bufferAddrCache = 0;
 
   rows.apply([&](vector_size_t i) {
     auto view = (uint32_t*)&utf8Views[2 * i];
```

Comment:
It should be okay to make this `reinterpret_cast<const uint32_t*>(&utf8Views[2 * i])`

Reply:
Ah well... that isn't sufficient as the view pointer is used to modify contents around line 811. It can't be const.

```
view[2] = bufferIdxCache;
view[3] = currAddr - bufferAddrCache;
```

Reply:
Can we do `const_cast<uint64_t*>(reinterpret_cast<const uint64_t*>(out.buffers[1]));`?

Reply:
@majetideepak : There is also guidance to not cast away const https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es50-dont-cast-away-const

I was leaning to add a temporary buffer for StringView value modification from Velox format -> Arrow format in VeloxToArrowBridgeHolder and then initialize this just once into the ArrowStruct  https://github.com/facebookincubator/velox/blob/main/velox/vector/arrow/Bridge.cpp#L43. That fits better with the class design.

But temporarily we can add const_cast if you are okay with that.

Reply:
The previous C-style cast is removing the const away anyway. I don't think this change will make anything worse.
The root cause seems to be that we use the same class for import and export.
The following field must then be.
`std::vector<const void*> buffers_{numBuffers_, nullptr};` -> `std::vector<void*> buffers_{numBuffers_, nullptr};`